launchable/test_runners/xctest.py

```diff
@@ -0,0 +1,69 @@
+import html
+import xml.etree.ElementTree as ET  # type: ignore
+
+import click
+from junitparser import TestCase, TestSuite
+
+from ..testpath import TestPath
+from . import launchable
+
+
+@click.argument('reports', nargs=-1, required=True)
+@launchable.record.tests
+def record_tests(client, reports):
+
+    def parse_func(p: str) -> ET.ElementTree:
+        tree = ET.parse(p)
+        root = tree.getroot()
+
+        for testsuite in root.findall('testsuite'):
+            for testcase in testsuite.findall('testcase'):
+                failure = testcase.find('failure')
+                if failure is not None:
+                    # Note(Konboi): XCTest escape `"` to `&quot;` in the message, so save as unescaped text
+                    message = html.unescape(failure.get('message', ''))
+                    body = failure.text or ''
+                    failure.text = f"{message}\n{body}"
+
+        return tree
+
+    def path_builder(case: TestCase, suite: TestSuite, report_path: str) -> TestPath:
+        class_attr = case.classname
+        splits = class_attr.split('.')
+        target = splits[0]
+        class_name = '/'.join(splits[1:])
```

Comment:
[Nit] You can write like this.
```suggestion
        [target, *rest] = class_attr.split('.')
        class_name = '/'.join(rest)
```